pom.xml

```diff
@@ -1605,6 +1605,260 @@
       </build>
     </profile>
 
+    <profile>
+      <id>jacoco-check</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.jacoco</groupId>
+            <artifactId>jacoco-maven-plugin</artifactId>
+            <version>${maven.jacoco.plugin.version}</version>
+            <executions>
+              <execution>
+                <id>default-instrument</id>
+                <goals>
+                  <goal>instrument</goal>
+                </goals>
+                <phase>test-compile</phase>
+              </execution>
+              <execution>
+                <id>default-restore-instrumented-classes</id>
+                <goals>
+                  <goal>restore-instrumented-classes</goal>
+                </goals>
+              </execution>
+              <execution>
+                <id>default-report</id>
+                <phase>site</phase>
+                <goals>
+                  <goal>report</goal>
+                </goals>
+              </execution>
+              <execution>
+                <id>default-check</id>
+                <goals>
+                  <goal>check</goal>
+                </goals>
+                <configuration>
+                  <dataFile>${project.build.directory}/jacoco/jacoco.exec</dataFile>
+                  <rules>
+                    <rule>
+                      <element>CLASS</element>
+                      <excludes>
+                        <!-- Terence Parr, The Definitive ANTLR 4 Reference, 2.4 Building Language Applications Using Parse Trees:
+                             "To better support access to the elements within specific nodes, ANTLR generates
+                             a RuleNode subclass for each rule. ... Given this description of the concrete types, we could write
+                             code by hand to perform a depth-first walk of the tree. We could perform whatever actions we
+                             wanted as we discovered and finished nodes. Typical operations are things such as
+                             computing results, updating data structures, or generating output."
```

Comment:
Why do we need this quote?
I see it is in the original cobertura, but why do we need it in either places?

Reply:
> why it's impossible to get 100% coverage for the ANTRL generated classes

I don't see how this comment does that. It just says how the class is generated.

Reply:
You're right. I've read the quote to the end - it does not say "impossible", it says "we do not need it". Well, it somehow explains why ANTLR classes do not have 100% coverage.